### PR TITLE
Character movement enabled, use keyboard arrows

### PR DESCRIPTION
**Pull request by abaddonpuff**
Character moves now! I added a couple helpers, Installed precommit with ruff for beautifullness (and cert ;) ) the movement mechanisms are now a class for the player with a flag on wether the arrow is pressed up or down.

### CODE REVIEW

File: sim/main_game.py [hunk 1]
```diff
@@ -3,40 +3,76 @@
...
+
+            # Make screen visible
+            self._update_screen()
+
+    def _check_events(self):
+        """
+        Responds to keypresses and mouse events
+        """
+
+        for event in pygame.event.get():
```
Comment (bbelderbos) [inline sim/main_game.py:44]: Can be shorter with dict + setattr:

```
key_map = {
    pygame.K_RIGHT: 'moving_right',
    pygame.K_LEFT: 'moving_left',
    pygame.K_UP: 'moving_up',
    pygame.K_DOWN: 'moving_down',
}

for event in pygame.event.get():
    if event.type == pygame.QUIT:
        sys.exit()
    elif event.type in (pygame.KEYDOWN, pygame.KEYUP):
        movement = event.type == pygame.KEYDOWN
        if event.key in key_map:
            setattr(self.player, key_map[event.key], movement)
```

(have not verified this yet)

File: sim/main_game.py [hunk 2]
```diff
@@ -3,40 +3,76 @@
...
+
+        pygame.display.flip()
 
-            #Make screen visible
-            self.screen.fill(self.settings.bg_color)
-            self.player.render()
-            pygame.display.flip()
 
-if __name__ == '__main__':
+if __name__ == "__main__":
```
Comment (bbelderbos) [inline sim/main_game.py:76]: ruff at work right? nice

File: characters/player/player.py [hunk 3]
```diff
@@ -1,28 +1,51 @@
 import pygame
 import os
 
-image_folder_path = os.path.join(os.path.dirname(os.path.dirname(__file__)), '..', 'images')
+image_folder_path = os.path.join(
```
Comment (bbelderbos) [inline characters/player/player.py]: changed to pathlib and pushed

File: README.md [hunk 4]
```diff
@@ -1,2 +1,9 @@
 # project_G
-Dating SIM using Python
\ No newline at end of file
+GAME!
+
+Instructions
+
+Run the game as:
+    python -m sim.main_game
```
Comment (bbelderbos) [inline README.md:7]: great

File: .pre-commit-config.yaml [hunk 5]
```diff
@@ -0,0 +1,18 @@
+repos:
```
Comment (bbelderbos) [inline .pre-commit-config.yaml:1]: nice

**Review by bbelderbos — APPROVED** [review]
awesome, now let's get some characters in the game and start conversation ...

File: sim/main_game.py (hunk 1 above)
Comment (abaddonpuff) [reply sim/main_game.py:44]: This works!!!

**Pull request merged**